lib/public/models/OverviewModel.js

```diff
@@ -119,8 +145,97 @@ export class OverviewPageModel extends Observable {
      * @return {Promise<void>} void
      */
     async load() {
-        const params = await this.getLoadParameters();
-        await this._dataSource.fetch(buildUrl(this.getRootEndpoint(), params));
+        const paginationParams = await this.getLoadParameters();
+        this._exportObservableItems.setCurrent(RemoteData.notAsked());
+        await this._dataSource.fetch(buildUrl(this.getRootEndpoint(), paginationParams));
+    }
+
+    /**
+     * Fetch all the relevant items from the API
+     * @return {Promise<void>} void
+     */
+    async loadExport() {
+        return this._exportObservableItems.getCurrent().match({
+            Success: () => null,
+            Loading: () => null,
+            Other: () => this._exportDataSource.fetch(this.getRootEndpoint()),
+        });
+    }
+
+    /**
+     * Create the export with the variables set in the model, handling errors appropriately
+     * @param {object[]} items The source content.
+     * @param {Object<string, Function<*, string>>} exportFormats defines how particual fields of data units will be formated
+     * @return {void}
+     */
+    async export(items, exportFormats = {}) {
+        const { exportFields, exportName } = this;
+        const exportData = items.map((item) => {
+            const entries = Object.entries(pick(item, exportFields));
+            const formattedEntries = entries.map(([key, value]) => {
+                const formatExport = exportFormats[key].exportFormat || ((identity) => identity);
+                return [key, formatExport(value, item)];
+            });
+            return Object.fromEntries(formattedEntries);
+        });
+        if (this.exportFormat === EXPORT_FORMATS.JSON) {
+            createJSONExport(exportData, `${exportName}.json`, 'application/json');
+        } else if (this.exportFormat === EXPORT_FORMATS.CSV) {
+            createCSVExport(exportData, `${exportName}.csv`, 'text/csv;charset=utf-8;');
+        } else {
+            throw new Error('Incorrect export format');
+        }
+    }
+
+    /**
+     * Get export name
+     * @return {string} name
+     */
+    get exportName() {
+        return this._exportName || this._defaultExportName;
+    }
+
+    /**
+     * Set export name
+     * @param {string} exportName name
+     */
+    set exportName(exportName) {
+        this._exportName = exportName;
+    }
+
+    /**
+     * Get the field values that will be exported
```

Comment:
I think this comment is misleading, because it seems like you return the value of the fields.
I would rather use `Get the list of properties that will be exported for each items`